src/buffer/out/search.cpp

```diff
@@ -13,23 +13,25 @@ bool Search::ResetIfStale(Microsoft::Console::Render::IRenderData& renderData, c
     const auto& textBuffer = renderData.GetTextBuffer();
     const auto lastMutationId = textBuffer.GetLastMutationId();
 
-    if (_needle == needle &&
+    if (_renderData == &renderData &&
+        _needle == needle &&

```

Comment:
I made this change because I grew a little worried about the robustness of the `_lastMutationId == lastMutationId` check.